internal/initwd/module_install.go

```diff
@@ -798,14 +818,23 @@ func (i *ModuleInstaller) installRegistryModule(ctx context.Context, req *config
 	if mod == nil {
 		// nil indicates missing or unreadable directory, so we'll
 		// discard the returned diags and return a more specific
-		// error message here. For registry modules this actually
-		// indicates a bug in the code above, since it's not the
-		// user's responsibility to create the directory in this case.
-		diags = diags.Append(&hcl.Diagnostic{
-			Severity: hcl.DiagError,
-			Summary:  "Unreadable module directory",
-			Detail:   fmt.Sprintf("The directory %s could not be read. This is a bug in OpenTofu and should be reported.", modDir),
-		})
+		// error message here.
+		if subDir != "" && isSubDirNonExistant(modDir) {
+			// This may be a user error, or a submodule may have been removed between unpinned versions of the module (ie a module update)
+			diags = diags.Append(&hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Module subdirectory not found",
+				Detail:   fmt.Sprintf("The subdirectory %q does not exist in module %q. Please verify that the subdirectory path is correct.", subDir, instPath),
```

Comment:
Question for reviewers, do you think its okay that i call it subdirectory here? instead of submodule? 

I was 50/50

Reply:
I think it's fine as is, it gets the point across.